File: user-service/src/main/java/com/phorus/userservice/UserServiceApplication.java
package com.phorus.userservice;

import com.phorus.userservice.config.CustomProperties;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;

@SpringBootApplication
@EnableConfigurationProperties(CustomProperties.class)
public class UserServiceApplication {

    public static void main(String[] args) {
        SpringApplication.run(UserServiceApplication.class, args);
    }

}

File: user-service/src/main/kotlin/com/phorus/userservice/UserServiceApplication.kt
package com.phorus.userservice

import com.phorus.userservice.config.CustomProperties
import io.r2dbc.spi.ConnectionFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.core.io.ClassPathResource
import org.springframework.r2dbc.connection.init.ConnectionFactoryInitializer
import org.springframework.r2dbc.connection.init.ResourceDatabasePopulator

@SpringBootApplication
@EnableConfigurationProperties(CustomProperties::class)
class UserServiceApplication {

	@Bean
	fun initializer(connectionFactory: ConnectionFactory?): ConnectionFactoryInitializer {
		val initializer = ConnectionFactoryInitializer()
		initializer.setConnectionFactory(connectionFactory!!)
		// This will create our database tables and schema
		initializer.setDatabasePopulator(ResourceDatabasePopulator(ClassPathResource("sql/schema.sql")))
		// This will drop our tables after we are done so we can have a fresh start next run
		initializer.setDatabaseCleaner(ResourceDatabasePopulator(ClassPathResource("sql/cleanup.sql")))
		return initializer
	}
}

fun main(args: Array<String>) {
	runApplication<UserServiceApplication>(*args)
}
